test/integration_download_test.go

```diff
@@ -0,0 +1,444 @@
+package test
```

Comment:
NOTE: This file is a refactoring of `integration_test.go` to make that main test file smaller. The only new test introduced here is `TestLocalDownloadWithAllowedHiddenFiles` - all other test functions are a straight copy.